ts/components/conversation/ModeratorsAddDialog.tsx

```diff
@@ -0,0 +1,217 @@
+  private readonly inputRef: React.RefObject<HTMLInputElement>;
+
+  constructor(props: any) {
+    super(props);
+
+    this.updateSearchBound = this.updateSearch.bind(this);
+    this.onMemberClicked = this.onMemberClicked.bind(this);
+    this.add = this.add.bind(this);
+    this.closeDialog = this.closeDialog.bind(this);
+    this.onClickOK = this.onClickOK.bind(this);
+    this.onKeyUp = this.onKeyUp.bind(this);
+    this.inputRef = React.createRef();
+
+    let friends = this.props.friendList;
+    friends = friends.map(d => {
+      const lokiProfile = d.getLokiProfile();
+      const name = lokiProfile ? lokiProfile.displayName : 'Anonymous';
+
+      // TODO: should take existing members into account
+      const existingMember = false;
+
+      return {
+        id: d.id,
+        authorPhoneNumber: d.id,
+        authorProfileName: name,
+        selected: false,
+        authorName: name,
+        authorColor: d.getColor(),
+        checkmarked: false,
+        existingMember,
+      };
+    });
+    this.state = {
+      friendList: friends,
+      inputBoxValue: '',
+    };
+
+    window.addEventListener('keyup', this.onKeyUp);
+  }
+
+  public updateSearch(event: React.FormEvent<HTMLInputElement>) {
+    const searchTerm = event.currentTarget.value;
+
+    const cleanedTerm = cleanSearchTerm(searchTerm);
+    if (!cleanedTerm) {
+      return;
+    }
+
+    this.setState(state => {
+      return {
+        ...state,
+        inputBoxValue: searchTerm,
+      };
+    });
+  }
+  public add() {
+    // if we have valid data
+    if (this.state.inputBoxValue.length > 64) {
+      const weHave = this.state.friendList.some(
+        user => user.authorPhoneNumber === this.state.inputBoxValue
+      );
+      if (!weHave) {
+        // lookup to verify it's registered?
+
+        // convert pubKey into local object...
+        const friends = this.state.friendList;
+        friends.push({
+          id: this.state.inputBoxValue,
+          authorPhoneNumber: this.state.inputBoxValue,
+          authorProfileName: this.state.inputBoxValue,
+          authorAvatarPath: '',
+          selected: true,
+          authorName: this.state.inputBoxValue,
+          authorColor: '#000000',
+          checkmarked: true,
+          existingMember: false,
+        });
+        this.setState(state => {
+          return {
+            ...state,
+            friendList: friends,
+          };
+        });
+      }
+      //
+    }
+    // clear
+    if (this.inputRef.current) {
+      this.inputRef.current.value = '';
+    }
+    this.setState(state => {
+      return {
+        ...state,
+        inputBoxValue: '',
+      };
+    });
+  }
+
+  public render() {
+    const i18n = window.i18n;
+    const titleText = `${i18n('addModerators')} ${this.props.chatName}`;
+
+    const hasFriends = this.state.friendList.length !== 0;
+
+    return (
+      <div className="content">
+        <p className="titleText">{titleText}</p>
+        Add Moderator:
```

Comment:
Should we use a translation here? Also, I would probably put this inside `<span>` or `<p>` so we could add CSS classes (but we can make it pretty later).